### DISCUSSION TITLE
detection training

### DISCUSSION THREAD

**Discussion by lfxuan** [mindee/doctr#736] [Q&A]
The data is synthesized and labeled in the same way, and the wrong image is not damaged after query. Why do the following errors appear in the training process
```
Traceback (most recent call last):
  File "references/detection/train_tensorflow.py", line 261, in <module>
    main(args)
  File "references/detection/train_tensorflow.py", line 200, in main
    fit_one_epoch(model, train_loader, batch_transforms, optimizer, mb)
  File "references/detection/train_tensorflow.py", line 42, in fit_one_epoch
    train_loss = model(images, targets, training=True)['loss']
  File "***/.pyenv/versions/doctr/lib/python3.7/site-packages/keras/engine/base_layer.py", line 1037, in __call__
    outputs = call_fn(inputs, *args, **kwargs)
  File "***/doctr/doctr/models/detection/differentiable_binarization/tensorflow.py", line 245, in call
    loss = self.compute_loss(logits, thresh_map, target)
  File "***/doctr/doctr/models/detection/differentiable_binarization/tensorflow.py", line 183, in compute_loss
    seg_target, seg_mask, thresh_target, thresh_mask = self.compute_target(target, out_map.shape[:3])
  File "***/doctr/doctr/models/detection/differentiable_binarization/base.py", line 326, in compute_target
    thresh_mask[idx])
  File "***/doctr/doctr/models/detection/differentiable_binarization/base.py", line 251, in draw_thresh_map
    canvas[ymin_valid:ymax_valid + 1, xmin_valid:xmax_valid + 1]
ValueError: could not broadcast input array from shape (3,0) into shape (3,1)
```

**Comment by fg-mindee** [comment]
Hi @lfxuan :wave: 

In order to help you, would you mind providing more details about how you got this error please? :pray: 
What do you mean by "the wrong image is not damaged after query" ? What arguments did you pass to the script?

Best

**Reply by lfxuan** [reply]
Hello! This is the image I made a mistake during training. I guess the coordinates of the image detection box appear negative after processing. What I do now is to change my image margin and try to avoid this situation to solve this problem.[errors.zip](https://github.com/mindee/doctr/files/7779719/errors.zip)

**Reply by fg-mindee** [reply]
Hi @lfxuan,

I think you forgot to upload the image :sweat_smile: 
I'm not sure I understand: did you manage to solve your problem? If yes, could you share how please ? :pray: If not, can you give more specifics please?

**Reply by fg-mindee** [reply]
Any update @lfxuan ? :pray:

**Reply by lfxuan** [reply]
@fg-mindee I'm very sorry to see the message now! Now the image and label files have been uploaded again
[error.zip](https://github.com/mindee/doctr/files/8138078/error.zip)
.

**Reply by fg-mindee** [reply]
It looks like a text recognition image. While your error seems to point out you were trying to train your model for text detection :thinking: 

Taking a look at your labels, it looks like you're trying to text a character-level localization model. The models in docTR are designed for word-level localization :sweat_smile: 

So a few options:
- if you are looking for text detection, character-level localization is fully experimental. We haven't tried it so far
- if you are looking for word-level localization, your image is already it. Considering a full page :)
